infrastructure/src/main/java/org/corfudb/infrastructure/management/ReconfigurationEventHandler.java

```diff
@@ -69,12 +68,26 @@ public static boolean handleFailure(@Nonnull IReconfigurationHandlerPolicy failu
      * @param healedServers     Set of healed server addresses
      * @param retryQueryTimeout Timeout to poll for workflow status.
      */
-    public static boolean handleHealing(@Nonnull IReconfigurationHandlerPolicy failureHandlerPolicy,
-                                        @Nonnull CorfuRuntime runtime,
+    public static boolean handleHealing(@Nonnull CorfuRuntime runtime,
                                         @Nonnull Set<String> healedServers,
                                         @Nonnull Duration retryQueryTimeout) {
         try {
             for (String healedServer : healedServers) {
+                // Try to estimate a reasonable timeout to rebuild the logging unit
+                Token trimMark = runtime.getAddressSpaceView().getTrimMark();
+                long tail = runtime.getAddressSpaceView().getAllTails().getLogTail();
+
+                long rangeToReplicate = tail - trimMark.getSequence();
+                // Since the orchestrator client and the fault detector client use
+                // the same configuration its reasonable to use these arguments.
+                // TODO(Maithem): AddNode should use a similar mechanisim to set the timeout
+                long numSections = rangeToReplicate / runtime.getParameters().getBulkReadSize();
+                long rpcTimeout = (long) (runtime.getParameters().getRequestTimeout().toMillis() * 1.5);
+                long timeoutInMs = Math.max(DEFAULT_HEAL_TIMEOUT.toMillis(), numSections * rpcTimeout);
```

Comment:
since we are making this dynamic based on number of sections, can we keep the lower bound to 2 minutes?